solr/core/src/java/org/apache/solr/update/CommitTracker.java

```diff
@@ -280,6 +282,16 @@ public void run() {
     } catch (Exception e) {
       log.error("auto commit error...", e);
     } finally {
+      if (timer != null) {
```

Comment:
QQ: have you considered metric for it?

Reply:
Yes 😊 I actually have another [branch](https://github.com/cowpaths/fullstory-solr/compare/fs/branch_9_3...cowpaths:fullstory-solr:patsonluk/commit-duration-metrics?expand=1) that did exactly that 

Unfortunately it didn't work too well:
1. Need to add metrics (as histogram) in the CommitTracker, and then convert it to grafana metrics in our MetricsServlet
2. We cannot simply report the histograms in our grafana metrics, since this will be per core and generate too many entries. Now if we combine it (as shown in the branch), then the result is not that accurate - since combining histogram is a lossy op
3. Our prometheus metrics servlet does not support histogram, so i picked mean and to account for weights of each core, i had to do some questionable math (get commit rate and use that as ratio), so the I was not too happy with the result.